velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -1159,6 +1162,89 @@ void ExpressionFuzzer::go() {
   logStats();
 }
 
+void ExpressionFuzzer::errorExit(const std::string& text) {
+  VELOX_FAIL(text);
+}
+
+void ExpressionFuzzer::drilldown(
+    core::TypedExprPtr plan,
+    const RowVectorPtr& rowVector,
+    const std::vector<column_index_t>& columnsToWrapInLazy) {
+  if (try2(plan, rowVector, columnsToWrapInLazy)) {
+    errorExit("Retry should have failed");
+  }
+  bool minimalFound = false;
+  drilldownRecursive(plan, rowVector, columnsToWrapInLazy, minimalFound);
+  if (minimalFound) {
+    errorExit("Found minimal failing expression");
+  } else {
+    errorExit("Only the top level expression failed");
+  }
+}
+
+void ExpressionFuzzer::drilldownRecursive(
+    core::TypedExprPtr plan,
+    const RowVectorPtr& rowVector,
+    const std::vector<column_index_t>& columnsToWrapInLazy,
+    bool& minimalFound) {
+  bool anyFailed = false;
+  for (auto& input : plan->inputs()) {
+    if (!try2(input, rowVector, columnsToWrapInLazy)) {
+      anyFailed = true;
+      drilldownRecursive(input, rowVector, columnsToWrapInLazy, minimalFound);
+      if (minimalFound) {
+        return;
+      }
+    }
+  }
+  if (!anyFailed) {
+    minimalFound = true;
+    LOG(INFO) << "Failed with all children succeeding: " << plan->toString();
```

Comment:
We should ideally now write this plan to File along with Vector so we can try debugging it - Many of these failures will occur during fuzzer runs on CI, so serializing here would really help.

Reply:
I was mistaken verify already does this, so the last verify should have the minimal repro.